dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracingInterceptor.kt

```diff
@@ -40,29 +42,60 @@ import okhttp3.Response
  */
 class TracingInterceptor : Interceptor {
 
+    private val localTracerReference: AtomicReference<Tracer> = AtomicReference()
```

Comment:
@xgouchet @ncreated what if we pass the tracer as a Constructor dependency and get rid of all this code ?

Reply:
I'm up for this idea 👍.

First, it introduces fully programmatic interface which cannot be broken semantically (say, I instantiate interceptor, but forget about registering tracer).

Second, it helps get rid of internal tracer, which now can be instantiated without further reasoning, but we don't know if in the future we won't need to have some initial state defined by the user.

Third, it seems just simpler - less code, less tests.

Reply:
I also agree...especially because of less code and the coder awareness

Reply:
Also what @xgouchet says make sense, I think in the end is better this way, better we control the things than the user.

Reply:
I see 🤔. I was thinking there's only one global tracer and users cannot create their own. Makes sense then 👍.